Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/OnCall.hs

```diff
@@ -0,0 +1,67 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Domain.Action.UI.OnCall
+  ( OnCallReq (..),
+    logOnCall,
+    sendCallDataToKafka,
+  )
+where
+
+import qualified Domain.Types.Ride as Ride
+import Kernel.External.Encryption
+import Kernel.Prelude
+import Kernel.Storage.Esqueleto.Config (EsqDBReplicaFlow)
+import Kernel.Storage.Esqueleto.Transactionable (runInReplica)
+import Kernel.Streaming.Kafka.Producer (produceMessage)
+import Kernel.Streaming.Kafka.Producer.Types (KafkaProducerTools)
+import qualified Kernel.Types.APISuccess as APISuccess
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import Storage.CachedQueries.CacheConfig (CacheFlow)
+import qualified Storage.Queries.Ride as QRide
+import Storage.Tabular.Person ()
+import Tools.Error
+
+newtype OnCallReq = OnCallReq
+  { rideId :: Id Ride.Ride
+  }
+  deriving (Generic, ToJSON, FromJSON, ToSchema)
+
+data ExotelRequestData = ExotelRequestData
+  { vendor :: Maybe Text,
+    time_stamp :: UTCTime,
+    ride_id :: Text,
+    call_type :: Text,
+    application :: Text,
+    call_sid :: Maybe Text,
+    status :: Maybe Text
+  }
```

Comment:
Lower camel casing should be used . Applies everywhere

Reply:
The data is required in snake case not camel case

Reply:
@PraveenGongada You can convert it into snakeCase via FromJson ToJson.
BTW why this data is required as snake_case ?

Reply:
@RitikaHanish Venkatesh asked me to send the data in snake_case